circuit-retained/src/androidInstrumentedTest/kotlin/com/slack/circuit/retained/android/RetainedTest.kt

```diff
@@ -696,15 +737,9 @@ private fun NestedRetainWithPushAndPopAndCannotRetain(useKeys: Boolean) {
     }
 
     // Keep the retained state registry around even if showNestedContent becomes false
-    CompositionLocalProvider(LocalCanRetainChecker provides CanRetainChecker.Always) {
+    CompositionLocalProvider(LocalCanRetainChecker provides { false }) {
```

Comment:
Is this line still necessary?

If `rememberRetained` has its state wiped, from leaving composition separately from either:
- being recreated
- the wrapping `RetainedStateProvider` leaving composition

Then I would also expect a `RetainedStateProvider` that leaves in a similar way to has its state wiped, I think.

Reply:
@alexvanyo 

The state of rememberRetained is wiped in two cases:
1.	When it itself leaves the composition.
2.	When the wrapping RetainedStateProvider leaves the composition.

In this case, the line `CompositionLocalProvider(LocalCanRetainChecker provides { false }) {` is still necessary, as it simulates a special situation where retaining is explicitly denied.

However, I realized a critical issue in my previous implementation based on your comment:
`Then I would also expect a RetainedStateProvider that leaves in a similar way to has its state wiped, I think.`

The problem is that my implementation did not differentiate between the case `when the RetainedStateProvider leaves the composition entirely` vs. `when its key changes`. As a result, the state of the RetainedStateProvider was not cleared when the RetainedStateProvider itself was removed from the composition.

To properly handle this, the RetainedStateProvider needs to distinguish between these two cases (key change vs. removal from composition). I believe this requires a modification similar to how rememberSaveableStateHolder provides a single holder.

I will make the necessary changes and update the code accordingly. Thanks!

Reply:
I fixed above in [these commits](https://github.com/vulpeszerda/circuit/compare/67d156f...9fbc2df)